File: content/examples/other/leetcode/problems/valid-anagram/example-2/solution.py
def is_anagram_counts(s, t):
    if len(s) != len(t):
        return False
    counts = [0] * 26
    for i in range(len(s)):
        counts[ord(s[i]) - ord("a")] += 1
        counts[ord(t[i]) - ord("a")] -= 1
    return all(x == 0 for x in counts)


if __name__ == "__main__":
    print(is_anagram_counts("anagram", "nagaram"))
    print(is_anagram_counts("rat", "car"))

File: content/examples/other/leetcode/problems/valid-anagram/example-2/Solution.java
public class Solution {
  static boolean isAnagramCounts(String s, String t) {
    if (s.length() != t.length()) return false;
    int[] counts = new int[26];
    for (int i = 0; i < s.length(); i += 1) {
      counts[s.charAt(i) - 'a'] += 1;
      counts[t.charAt(i) - 'a'] -= 1;
    }
    for (int x : counts) if (x != 0) return false;
    return true;
  }

  public static void main(String[] args) {
    System.out.println(isAnagramCounts("anagram", "nagaram"));
    System.out.println(isAnagramCounts("rat", "car"));
  }
}
